core/trino-main/src/main/java/io/trino/execution/DataDefinitionExecution.java

```diff
@@ -219,6 +220,12 @@ public void fail(Throwable cause)
         stateMachine.transitionToFailed(cause);
     }
 
+    @Override
+    public boolean isPlanningFinished()
+    {
+        return getState().ordinal() > PLANNING.ordinal();
```

Comment:
There are 4 methods that look exactly like that. One way of fixing that is introducing `getState` method to `TrackedQuery` interface.

Reply:
Add method to `QueryState` (`isPlanningFinished`). I would also use explicit boolean in `QueryState` (`planningFinished`) or explicit case `io.trino.execution.StageState#canScheduleMoreTasks`.